contrib/common.cpp

```diff
@@ -95,3 +95,47 @@ void onedim_fseries_kernel(BIGINT nf, FLT *fwkerhalf, SPREAD_OPTS opts)
     }
   }
 }
+
+void onedim_fseries_kernel_1sthalf(BIGINT nf, FLT *f, dcomplex *a, SPREAD_OPTS opts)
+{
+  FLT J2 = opts.nspread/2.0;            // J/2, half-width of ker z-support
+  // # quadr nodes in z (from 0 to J/2; reflections will be added)...
+  int q=(int)(2 + 3.0*J2);  // not sure why so large? cannot exceed MAX_NQUAD
+  double z[2*MAX_NQUAD],w[2*MAX_NQUAD];
+  legendre_compute_glr(2*q,z,w);        // only half the nodes used, eg on (0,1)
+  for (int n=0;n<q;++n) {               // set up nodes z_n and vals f_n
+    z[n] *= J2;                         // rescale nodes
+    f[n] = J2*(FLT)w[n] * evaluate_kernel((FLT)z[n], opts); // vals & quadr wei
+    a[n] = exp(2*PI*IMA*(FLT)(nf/2-z[n])/(FLT)nf);  // phase winding rates
+  }
+}
+
+#if 0
```

Comment:
Might make sense to uncomment this and have `onedim_fseries_kernel` call both functions. This way we make sure that the same code is being used in the full CPU and the CPU/GPU hybrid cases.

Also, the name here could be `onedim_fseries_kernel_compute` or something to avoid the first half/second half terminology.